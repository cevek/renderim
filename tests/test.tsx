import {createElement, Suspense} from 'renderim';
import {createLoadData, render, getNextRestartedComponent} from './base';

test('Suspense mount', done => {
    const LoadData = createLoadData();
    const rootNode = render(
        <Suspense timeout={50} fallback={<div>Loading...</div>}>
            <div class="wrapper">
                <LoadData ms={20} />
                Hello
            </div>
        </Suspense>,
    );
    expect(rootNode).toMatchSnapshot('root');
    setTimeout(() => {
        expect(getNextRestartedComponent()).toMatchSnapshot('Suspense');
        expect(getNextRestartedComponent()).toMatchSnapshot('Data');
        done();
    }, 100);
});
