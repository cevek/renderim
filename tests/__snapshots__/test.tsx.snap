// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suspense mount: Data 1`] = `
"<Suspense>
   <Fragment>
      <div>
         <Data>
            data
         </Data>
         Hello
      </div>
   </Fragment>
</Suspense>"
`;

exports[`Suspense mount: Suspense 1`] = `
"<Data>
   data
</Data>"
`;

exports[`Suspense mount: root 1`] = `
"<Suspense>
   <Fragment>
      <div>
         <Data/>
         Hello
      </div>
   </Fragment>
</Suspense>"
`;
